### PR TITLE
feat: export KongAPIError and expose HTTP status code

### LINKED ISSUE #10: Expose HTTP StatusCode for kongAPIError
HTTP Status Code is embedded in the error string. There are times when the caller would like to know the error code and act accordingly (such as https://github.com/Kong/deck/pull/226).

https://github.com/Kong/go-kong/blob/main/kong/error.go

### PR DESCRIPTION
**Pull request by krijohs**
Exports KongAPIError and a method `Code` is added to it which returns the HTTP status code

Closes #10

### CODE REVIEW

**Comment by hbagdi** [comment]
Please sign the CLA @krijohs, thanks!

**Comment by hbagdi** [comment]
Enterprise failures can be ignored, the CI borks for them on PRs.

File: kong/error.go [hunk 1]
```diff
@@ -4,20 +4,24 @@ import (
 	"fmt"
 )
 
-type kongAPIError struct {
+type APIError struct {
```
Comment (hbagdi) [inline kong/error.go:7]: Can you add a doc line to this type and the functions?
This is required because this type is now exported.

Comment (krijohs) [reply kong/error.go:7]: Sure, will add it.

File: kong/error.go [hunk 2]
```diff
@@ -4,20 +4,26 @@ import (
...
+func (e *APIError) Code() int {
+	return e.httpCode
+}
+
 // IsNotFoundErr returns true if the error or it's cause is
 // a 404 response from Kong.
 func IsNotFoundErr(e error) bool {
 	switch e := e.(type) {
-	case *kongAPIError:
+	case *APIError:
```
Comment (hbagdi) [inline kong/error.go:26]: This function can be deprecated at this point.

**Review by hbagdi — APPROVED** [review]

**Comment by hbagdi** [comment]
Thank you for your contribution. Please fill out the following form to claim your contributor swag:
https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#contributor-t-shirt.

File: kong/error.go (hunk 2 above)
Comment (krijohs) [reply kong/error.go:26]: I thought about it but wasn't sure if you wanted to keep it for backwards compability.

**Pull request merged**